File: java/spring/base-project/src/main/java/com/github/mroose/backend/controller/SampleController.java
package com.github.mroose.backend.controller;

import com.github.mroose.backend.dto.SampleDto;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.bind.annotation.GetMapping;
import com.github.mroose.backend.service.SampleService;
import reactor.core.publisher.Flux;

@RestController
@RequiredArgsConstructor
public class SampleController {

    private final SampleService sampleService;

    @GetMapping("/api/v1/todos")
    public Flux<SampleDto> getTodos() {
        return sampleService.getTodos();
    }
}

File: kotlin/spring/base-project/src/main/kotlin/com/github/mroose/backend/controller/SampleController.kt
package com.github.mroose.backend.controller

import com.github.mroose.backend.dto.SampleDto
import com.github.mroose.backend.service.SampleService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController
import reactor.core.publisher.Flux

@RestController
class SampleController(val sampleService: SampleService) {

    @GetMapping("/api/v1/todos")
    fun getTodos(): Flux<SampleDto> {
        return sampleService.getTodos()
    }
}